gm4_blossoming_pots/data/gm4_blossoming_pots/function/main.mcfunction

```diff
@@ -0,0 +1,5 @@
+execute as @e[tag=gm4_blossoming_pots.display.decorated_pot] at @s unless block ~ ~ ~ minecraft:decorated_pot run kill @s
+execute as @e[tag=gm4_blossoming_pots.display.flower_pot] at @s unless block ~ ~ ~ minecraft:flower_pot run kill @s
+execute as @e[tag=gm4_blossoming_pots.data.flower_pot] at @s unless block ~ ~ ~ minecraft:flower_pot run function gm4_blossoming_pots:flower/item/break with entity @s data
```

Comment:
These `@e` checks should have a `type=`. But also I feel like this could be a single main command with a shared `tag=` and a block tag, running a cleanup function that does more work.